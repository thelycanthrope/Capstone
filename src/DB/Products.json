{
  "products": [
    {
      "pname": "SS Master 1000 English Willow Cricket Bat",
      "description": " Lorem ipsum dolor sit amet consectetur adipisicing elit. Dicta magni et nemo, voluptate sed cum voluptates voluptatum provident",
      "manufacturer": "Sareen Sports Industries",
      "price": 6300,
      "quantity": "78",
      "url": "https://www.onlinecricstore.com/media/catalog/product/cache/26c91cbb37c57611b1daf32fd10f6d8a/u/n/untitled-1_25.png",
      "id": 1,
      "category": "bat"
    },
    {
      "pname": "Kookaburra Kahuna 750 Wicket Keeping Gloves",
      "quantity": 45,
      "price": 2900,
      "url": "https://www.onlinecricstore.com/media/catalog/product/cache/26c91cbb37c57611b1daf32fd10f6d8a/u/n/untitled-1_11_1.png",
      "description": " Lorem ipsum dolor sit amet consectetur adipisicing elit. Dicta magni et nemo, voluptate sed cum voluptates voluptatum provident",
      "manufacturer": "Kookaburra Sports",
      "id": 2,
      "category": "gloves"
    },
    {
      "pname": "SG Club White Leather Cricket Ball",
      "quantity": 25,
      "price": 900,
      "url": "https://www.onlinecricstore.com/media/catalog/product/cache/26c91cbb37c57611b1daf32fd10f6d8a/c/l/club-white3.jpg",
      "description": " Lorem ipsum dolor sit amet consectetur adipisicing elit. Dicta magni et nemo, voluptate sed cum voluptates voluptatum provident",
      "manufacturer": "SG SPorts",
      "id": 3,
      "category": "ball"
    },
    {
      "pname": "Ceat Grippstar Cricket Batting Pads",
      "quantity": 18,
      "price": 4000,
      "url": "https://www.onlinecricstore.com/media/catalog/product/cache/26c91cbb37c57611b1daf32fd10f6d8a/d/s/dsc_0395_4.png",
      "description": " Lorem ipsum dolor sit amet consectetur adipisicing elit. Dicta magni et nemo, voluptate sed cum voluptates voluptatum provident",
      "manufacturer": "CEAT Sports",
      "id": 4,
      "category": "pads"
    },
    {
      "pname": "MRF Pro Bails - Senior",
      "quantity": 10,
      "price": 200,
      "url": "https://cdn-5c84bc36-b681cbc1.mysagestore.com/b522fd52e101edc926c3308c230445d5/contents/3T163225/thumbnail/middle_3T163225.jpg",
      "description": " Lorem ipsum dolor sit amet consectetur adipisicing elit. Dicta magni et nemo, voluptate sed cum voluptates voluptatum provident",
      "manufacturer": "MRF Sports",
      "id": 5,
      "category": "accessories"
    },
    {
      "pname": "Kookaburra Kahuna 1000 English Willow Cricket Bat",
      "quantity": 8,
      "price": 21000,
      "url": "https://www.onlinecricstore.com/media/catalog/product/cache/26c91cbb37c57611b1daf32fd10f6d8a/k/a/kahuna-1000-4_1.png",
      "description": " Lorem ipsum dolor sit amet consectetur adipisicing elit. Dicta magni et nemo, voluptate sed cum voluptates voluptatum provident",
      "manufacturer": "Kookaburra Sports",
      "id": 6,
      "category": "bat"
    },
    {
      "pname": "MRF Genius Grand Edition Cricket Bat English Willow",
      "quantity": 5,
      "price": 8500,
      "url": "https://www.onlinecricstore.com/media/catalog/product/cache/26c91cbb37c57611b1daf32fd10f6d8a/c/r/cricketbat_e_2_1.png",
      "description": " Lorem ipsum dolor sit amet consectetur adipisicing elit. Dicta magni et nemo, voluptate sed cum voluptates voluptatum provident ",
      "manufacturer": "MRF Sports",
      "id": 7,
      "category": "bat"
    },
    {
      "pname": "Kookaburra Surge 200 English Willow Cricket Bat",
      "quantity": 8,
      "price": 9900,
      "url": "https://www.onlinecricstore.com/media/catalog/product/cache/26c91cbb37c57611b1daf32fd10f6d8a/u/n/untitled-1_69.png",
      "description": " Lorem ipsum dolor sit amet consectetur adipisicing elit. Dicta magni et nemo, voluptate sed cum voluptates voluptatum provident",
      "manufacturer": "Kookaburra Sports",
      "id": 8,
      "category": "bat"
    },
    {
      "pname": "SG PLAYER ULTIMATE CRICKET BAT ENGLISH WILLOW",
      "quantity": 5,
      "price": 13000,
      "url": "https://www.onlinecricstore.com/media/catalog/product/cache/26c91cbb37c57611b1daf32fd10f6d8a/u/n/untitled-1_1_1.png",
      "description": " Lorem ipsum dolor sit amet consectetur adipisicing elit. Dicta magni et nemo, voluptate sed cum voluptates voluptatum provident ",
      "manufacturer": "SG Sports",
      "id": 9,
      "category": "bat"
    },
    {
      "pname": "BAS VAMPIRE Commander English Willow Cricket Bat",
      "quantity": 4,
      "price": 11000,
      "url": "https://www.onlinecricstore.com/media/catalog/product/cache/26c91cbb37c57611b1daf32fd10f6d8a/u/n/untitled-1_79_1.png",
      "description": " Lorem ipsum dolor sit amet consectetur adipisicing elit. Dicta magni et nemo, voluptate sed cum voluptates voluptatum provident",
      "manufacturer": "BAS Vampire",
      "id": 10,
      "category": "bat"
    },
    {
      "pname": "DSC Condor Drive English Willow Cricket Bat",
      "quantity": 8,
      "price": 7000,
      "url": "https://www.onlinecricstore.com/media/catalog/product/cache/26c91cbb37c57611b1daf32fd10f6d8a/d/s/dsc_05080.png",
      "description": " Lorem ipsum dolor sit amet consectetur adipisicing elit. Dicta magni et nemo, voluptate sed cum voluptates voluptatum provident",
      "manufacturer": "DSC Sports",
      "id": 11,
      "category": "bat"
    },
    {
      "pname": "SG Hybrid 20 Xtreme English Willow Cricket Bat",
      "quantity": 5,
      "price": 14000,
      "url": "https://www.onlinecricstore.com/media/catalog/product/cache/26c91cbb37c57611b1daf32fd10f6d8a/s/g/sg-hybrid-20-xtreme-mult-image.png",
      "description": " Lorem ipsum dolor sit amet consectetur adipisicing elit. Dicta magni et nemo, voluptate sed cum voluptates voluptatum provident ",
      "manufacturer": "SG Sports",
      "id": 12,
      "category": "bat"
    },
    {
      "pname": "SS Heritage English Willow Cricket Bat",
      "quantity": 4,
      "price": 6500,
      "url": "https://www.onlinecricstore.com/media/catalog/product/cache/26c91cbb37c57611b1daf32fd10f6d8a/1/2/123_2.png",
      "description": " Lorem ipsum dolor sit amet consectetur adipisicing elit. Dicta magni et nemo, voluptate sed cum voluptates voluptatum provident",
      "manufacturer": "Sareen Sports Industries",
      "id": 13,
      "category": "bat"
    },
    {
      "pname": "SS Gladiator Cricket Batting Gloves",
      "quantity": 14,
      "price": 3000,
      "url": "https://www.onlinecricstore.com/media/catalog/product/cache/26c91cbb37c57611b1daf32fd10f6d8a/5/_/5_103_1.png",
      "description": " Lorem ipsum dolor sit amet consectetur adipisicing elit. Dicta magni et nemo, voluptate sed cum voluptates voluptatum provident",
      "manufacturer": "Sareen Sports Industries",
      "id": 14,
      "category": "gloves"
    },
    {
      "pname": "SG RSD Supalite Cricket Batting Gloves",
      "quantity": 12,
      "price": 2500,
      "url": "https://www.onlinecricstore.com/media/catalog/product/cache/26c91cbb37c57611b1daf32fd10f6d8a/u/n/untitled-1_39.png",
      "description": " Lorem ipsum dolor sit amet consectetur adipisicing elit. Dicta magni et nemo, voluptate sed cum voluptates voluptatum provident",
      "manufacturer": "SG Sports",
      "id": 15,
      "category": "gloves"
    },
    {
      "pname": "Spartan Camaflauge Wicket Keeping Gloves",
      "quantity": 4,
      "price": 1800,
      "url": "https://www.onlinecricstore.com/media/catalog/product/cache/26c91cbb37c57611b1daf32fd10f6d8a/u/n/untitled-1_69_1.png",
      "description": " Lorem ipsum dolor sit amet consectetur adipisicing elit. Dicta magni et nemo, voluptate sed cum voluptates voluptatum provident",
      "manufacturer": "Spartan Sports ",
      "id": 16,
      "category": "gloves"
    },
    {
      "pname": "SG Club Leather Cricket Red Ball",
      "quantity": 36,
      "price": 1500,
      "url": "https://www.onlinecricstore.com/media/catalog/product/cache/26c91cbb37c57611b1daf32fd10f6d8a/c/l/club2.jpg",
      "description": " Lorem ipsum dolor sit amet consectetur adipisicing elit. Dicta magni et nemo, voluptate sed cum voluptates voluptatum provident",
      "manufacturer": "SG Sports",
      "id": 17,
      "category": "ball"
    },
    {
      "pname": "SS Gladiator Cricket Batting Pads",
      "quantity": 16,
      "price": 3500,
      "url": "https://www.onlinecricstore.com/media/catalog/product/cache/26c91cbb37c57611b1daf32fd10f6d8a/6/5/65.png",
      "description": " Lorem ipsum dolor sit amet consectetur adipisicing elit. Dicta magni et nemo, voluptate sed cum voluptates voluptatum provident",
      "manufacturer": "Sareen Sports Industries",
      "id": 18,
      "category": "pads"
    },
    {
      "pname": "Kookaburra Elite Cricket Helmet",
      "quantity": 24,
      "price": 1200,
      "url": "https://cdn-5c84bc36-b681cbc1.mysagestore.com/b522fd52e101edc926c3308c230445d5/contents/3M183/thumbnail/middle_3M183.jpg",
      "description": " Lorem ipsum dolor sit amet consectetur adipisicing elit. Dicta magni et nemo, voluptate sed cum voluptates voluptatum provident",
      "manufacturer": "Kookaburra Sports",
      "id": 19,
      "category": "accessories"
    },
    {
      "pname": "Savage Player Edition Cricket Bat Grip, Pack of 3",
      "description": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Dicta magni et nemo, voluptate sed cum voluptates voluptatum provident",
      "manufacturer": "NB Sports",
      "price": "299",
      "quantity": "18",
      "url": "https://www.onlinecricstore.com/media/catalog/product/cache/26c91cbb37c57611b1daf32fd10f6d8a/u/n/untitled-1_71.png",
      "category": "accessories",
      "id": 20
    },
    {
      "pname": "SS Master 1000 English Willow Cricket Bat",
      "description": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Dicta magni et nemo, voluptate sed cum voluptates voluptatum provident",
      "manufacturer": "Sareen Sports Industries",
      "price": "6300",
      "quantity": "55",
      "url": "https://www.onlinecricstore.com/media/catalog/product/cache/26c91cbb37c57611b1daf32fd10f6d8a/u/n/untitled-1_25.png",
      "category": "bat",
      "id": 21
    }
  ],
  "users": [
    {
      "email": "example1@email.com",
      "password": "password@321",
      "fname": "Alaric",
      "lname": "Saltzman",
      "location": "Mystic falls",
      "contactno": "+324568423794",
      "id": 1
    },
    {
      "email": "example2@email.com",
      "password": "password#132",
      "fname": "Stiles",
      "lname": "Stilinski",
      "location": "Beackon Hills",
      "contactno": "+365559557",
      "id": 2
    },
    {
      "email": "example3@email.com",
      "password": "password$432",
      "fname": "Niklaus",
      "lname": "Mikaelson",
      "location": "New Orleans",
      "contactno": "+16135550119",
      "id": 3
    },
    {
      "email": "example4@email.com",
      "password": "password%145",
      "fname": "Malcom",
      "lname": "Merlin",
      "location": "Starling City",
      "contactno": "+3854573136810",
      "id": 4
    },
    {
      "email": "exampl2@email.co.in",
      "password": "123123",
      "fname": "exampl",
      "lname": "2",
      "location": "solapur",
      "contactno": "1234569870",
      "id": 5
    },
    {
      "email": "abcd@email.com",
      "password": "abcdabcd",
      "fname": "Jarvis",
      "lname": "S",
      "location": "NYC",
      "contactno": "+456789123",
      "id": 6
    },
    {
      "email": "xyz@email.com",
      "password": "asdfghjkl",
      "fname": "King",
      "lname": "K",
      "location": "Tahiti",
      "contactno": "+789456123",
      "id": 7
    }
  ],
  
  "views": [
    {
      "bat": 0,
      "ball": 0,
      "pads": 0,
      "gloves": 0,
      "accessories": 0,
      "id": 1
    }
  ]
}