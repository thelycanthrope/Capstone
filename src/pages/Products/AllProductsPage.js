import React,{ useEffect, useState,Suspense,lazy} from 'react'
import fetchData from '../../Actions/actions'
import {useSelector,useDispatch} from 'react-redux'
import ReactTooltip from 'react-tooltip';
import { Link} from 'react-router-dom';
import {HashLoader} from 'react-spinners'
import 'react-dropdown/style.css';
import { Field, Form, Formik } from 'formik';
import NavBar from '../../Components/NavBar';
const ProductList=lazy( () => import('./ProductList'))






function AllProductsPage(props) {
    const products= useSelector( (state) => {return state});
    const [flag,setFlag]=useState(false)
    const [array,setArray]=useState([])
    console.log("all prod page",products);
    const dispatch = useDispatch();
  
    const [is,setIs]=useState();
    useEffect( () => {
    //     axios.get('http://localhost:5500/loggedin/1')
    // .then( r => { setIs(r.data.loggedin) })
    // .catch( e => console.log(e))
    dispatch(fetchData())
    setIs(localStorage.getItem('loggedin'))

},[])

  
    const handleSubmit = (values) => {
        
        if(values.options!=='All'){
           setFlag(true)
        setArray( products.products.filter( (prod) =>{
              return (prod.category===values.options.toLowerCase());
        }))}
        else if (values.options==='All'){
           setFlag(false)
        }

    
    }




    return (
        
            <div style={{ backgroundColor: "black" }}>
                
                <NavBar/>
         
                
                 <Formik  initialValues={{category:'options'}} onSubmit={ handleSubmit.bind(products)}>

                    <Form >
                <Field as='select'  name="options" className="bg-dark text-white mr-3 h-100"  data-tip data-for="category">
                <option value="All">All</option>
               
                    <option value="Bat" >Bat</option>
                    <option value="Ball">Ball</option>
                    <option value="Pads">Pads</option>
                    <option value="Gloves">Gloves</option>
                    <option value="Accessories">Accessories</option>
                </Field>
                            <ReactTooltip id="category" place="top" effect="solid" >
                                Select category
                            </ReactTooltip>
                <button type="submit" className=" btn btn-dark text-warning"> Filter</button>
                {is
                        ?<> <Link to="/addproduct" className="btn btn-dark btn-outline-light  float-right"  data-tip data-for="addd">Add Product</Link>
                        <ReactTooltip id="addd" place="bottom" effect="solid" >
                                Add Product
                            </ReactTooltip></>

                        : <>
                            <button className="btn btn-dark btn-outline-light  float-right" data-tip data-for="add"> Add Product</button>
                            <ReactTooltip id="add" place="bottom" effect="solid" >
                                Please Sign in to add
                            </ReactTooltip>
                        </>
                }
              
                </Form>
                </Formik>
                
                
                  
           
            <Suspense fallback={<div className="text-center"><br/><br/><br/><br/><br/><br/><br/><br/><br/><br/><br/><br/><br/><br/><br/><HashLoader size={140} color='orange'/><br/><br/><br/><br/><br/><br/><br/><br/><br/><br/><br/><br/><br/><br/><br/></div>}>
                    
               
                {flag===true
           ?<ProductList products={array}/>
            :typeof(products)!=='undefined'?
            <ProductList products={products.products}/>:null
            }
           
           </Suspense>
           
        </div>
    )
}
export default (AllProductsPage)




