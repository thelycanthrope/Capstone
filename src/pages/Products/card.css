.carddd:hover{
    box-shadow: rgb(206 199 199 / 20%) 0px 0px 19px 11px;
}

h5:hover{
    cursor: pointer;
}