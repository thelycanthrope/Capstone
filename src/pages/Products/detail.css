img{
    height:600px;
    width: 500px;
}
.blink{
    width:200px;
    height: 50px;
    background-color: #ff9400;
    padding: 15px;	
    text-align: center;
    line-height: 50px;
}
span.blink{
    font-size: 25px;
    font-family: cursive;
    color: white;
    animation: blink 1s linear infinite;
}
@keyframes blink {
    0%{opacity: 0;}
50%{opacity: .5;}
100%{opacity: 1;}
    
}