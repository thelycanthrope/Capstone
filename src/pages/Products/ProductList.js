import React from 'react'
import Products from "./Products";

function ProductList(props) {
   

    let node = props.products && props.products.map(prod=> (
        
        <Products id={prod.id}
            key={prod.id}
            pname={prod.pname}
            price={prod.price} 
            url={prod.url}
            description={prod.description}
            quantity={prod.quantity}
            manufacturer={prod.manufacturer}
            category={prod.category}
            />
        
    ))
   
    
    return (
        <div>
            <div className="container " >
           
                <div className="row">
                    {node}
                </div>
            </div>
        </div>
    )
}

export default ProductList
