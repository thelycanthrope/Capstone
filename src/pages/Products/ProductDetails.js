import axios from 'axios';
import React, { useEffect, useState } from 'react'
import { useHistory, useLocation, withRouter} from 'react-router-dom'
import {Link} from 'react-router-dom';
import ReactTooltip from 'react-tooltip';
import {toast} from 'react-toastify'
import 'react-toastify/dist/ReactToastify.css'
import './detail.css'
import NavBar from '../../Components/NavBar';


let history;

toast.configure();

function ProductDetails() {
    window.scrollTo(0,0)
   const location=useLocation()
    const [is,setIs]=useState();
    const [topViewed, settopViewed] = useState()
 
    
const category=location.state.category
    
   
   history=useHistory();


    useEffect( () => {
        
   

    setIs(localStorage.getItem('loggedin'))
 
    axios.get('http://localhost:5500/views/1')
.then( (r)=>{
   
     

settopViewed({...r.data})



        switch(category){
    case 'bat':
    topViewed.bat=topViewed.bat+1
        break;

    case 'ball':
        
        topViewed.ball=topViewed.ball+1
 
     break;

    case 'pads':
     
        topViewed.pads=topViewed.pads+1
       
    break;

    case 'gloves':

        topViewed.gloves=topViewed.gloves+1

    break;

    case 'accessories':

        topViewed.accessories=topViewed.accessories+1
    break;

    default:console.log(category);
    
}

topViewed?
axios.put('http://localhost:5500/views/1',topViewed):null
.then(r=>console.log(r))
.catch(e => console.log(e))

}

)
.catch( e => console.log(e))





},[category,topViewed])
 
    return (
        
        <div className="bg-dark text-white">
            <NavBar/>

            {is
         ? <><Link to={{pathname:'/update',state:location}}><svg xmlns="http://www.w3.org/2000/svg" width="35" height="35" fill="currentColor" className="bi bi-pencil border mt-2 ml-2 p-1" viewBox="0 0 16 16" data-tip data-for="edit">
                <path d="M12.146.146a.5.5 0 0 1 .708 0l3 3a.5.5 0 0 1 0 .708l-10 10a.5.5 0 0 1-.168.11l-5 2a.5.5 0 0 1-.65-.65l2-5a.5.5 0 0 1 .11-.168l10-10zM11.207 2.5 13.5 4.793 14.793 3.5 12.5 1.207 11.207 2.5zm1.586 3L10.5 3.207 4 9.707V10h.5a.5.5 0 0 1 .5.5v.5h.5a.5.5 0 0 1 .5.5v.5h.293l6.5-6.5zm-9.761 5.175-.106.106-1.528 3.821 3.821-1.528.106-.106A.5.5 0 0 1 5 12.5V12h-.5a.5.5 0 0 1-.5-.5V11h-.5a.5.5 0 0 1-.468-.325z" />
            </svg></Link> 
            <ReactTooltip id="edit" place="bottom" effect="solid">
                Update Product
            </ReactTooltip> </> 
            : <>
            <svg xmlns="http://www.w3.org/2000/svg" width="35" height="35" fill="currentColor" className="bi bi-pencil border mt-2 ml-2 p-1" viewBox="0 0 16 16" data-tip data-for="edit">
                <path d="M12.146.146a.5.5 0 0 1 .708 0l3 3a.5.5 0 0 1 0 .708l-10 10a.5.5 0 0 1-.168.11l-5 2a.5.5 0 0 1-.65-.65l2-5a.5.5 0 0 1 .11-.168l10-10zM11.207 2.5 13.5 4.793 14.793 3.5 12.5 1.207 11.207 2.5zm1.586 3L10.5 3.207 4 9.707V10h.5a.5.5 0 0 1 .5.5v.5h.5a.5.5 0 0 1 .5.5v.5h.293l6.5-6.5zm-9.761 5.175-.106.106-1.528 3.821 3.821-1.528.106-.106A.5.5 0 0 1 5 12.5V12h-.5a.5.5 0 0 1-.5-.5V11h-.5a.5.5 0 0 1-.468-.325z" />
            </svg>
            <ReactTooltip id="edit" place="bottom" effect="solid">
                Please sign in to Update Product
            </ReactTooltip> </> }


            {is
                ?<button className="btn btn-dark bg-dark float-right border-0 pt-0 pr-0" onClick={delFun.bind(location.state.id)}>
            <svg xmlns="http://www.w3.org/2000/svg" width="35" height="35" fill="currentColor" className="bi bi-trash border  mt-2 mr-2 p-1" viewBox="0 0 16 16" data-tip data-for="delete"   >
                <path d="M5.5 5.5A.5.5 0 0 1 6 6v6a.5.5 0 0 1-1 0V6a.5.5 0 0 1 .5-.5zm2.5 0a.5.5 0 0 1 .5.5v6a.5.5 0 0 1-1 0V6a.5.5 0 0 1 .5-.5zm3 .5a.5.5 0 0 0-1 0v6a.5.5 0 0 0 1 0V6z" />
                <path fillRule="evenodd" d="M14.5 3a1 1 0 0 1-1 1H13v9a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V4h-.5a1 1 0 0 1-1-1V2a1 1 0 0 1 1-1H6a1 1 0 0 1 1-1h2a1 1 0 0 1 1 1h3.5a1 1 0 0 1 1 1v1zM4.118 4 4 4.059V13a1 1 0 0 0 1 1h6a1 1 0 0 0 1-1V4.059L11.882 4H4.118zM2.5 3V2h11v1h-11z" />
            </svg>
            <ReactTooltip id="delete" place="bottom" effect="solid">
                Delete Product
            </ReactTooltip>
            </button>

            :<>
            <svg xmlns="http://www.w3.org/2000/svg" width="35" height="35" fill="currentColor" className="bi bi-trash border float-right mt-2 mr-2 p-1" viewBox="0 0 16 16" data-tip data-for="delete"   >
                <path d="M5.5 5.5A.5.5 0 0 1 6 6v6a.5.5 0 0 1-1 0V6a.5.5 0 0 1 .5-.5zm2.5 0a.5.5 0 0 1 .5.5v6a.5.5 0 0 1-1 0V6a.5.5 0 0 1 .5-.5zm3 .5a.5.5 0 0 0-1 0v6a.5.5 0 0 0 1 0V6z" />
                <path fillRule="evenodd" d="M14.5 3a1 1 0 0 1-1 1H13v9a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V4h-.5a1 1 0 0 1-1-1V2a1 1 0 0 1 1-1H6a1 1 0 0 1 1-1h2a1 1 0 0 1 1 1h3.5a1 1 0 0 1 1 1v1zM4.118 4 4 4.059V13a1 1 0 0 0 1 1h6a1 1 0 0 0 1-1V4.059L11.882 4H4.118zM2.5 3V2h11v1h-11z" />
            </svg>
            <ReactTooltip id="delete" place="bottom" effect="solid">
                Please sign in Delete Product
            </ReactTooltip>
            </>}

            <br></br>
            <div className="container bg-dark text-white ">
                <div className="row">
                    <div className="col sm">
                    <img src={location.state.url} className="border img-thumbnail rounded mx-3 my-5 bg-dark" alt=""></img>
                    </div>
                    <div className="col sm">
                    <h1>{location.state.pname}</h1>
                    <h2 className="text-center"> <span className="blink">&#8377; {location.state.price} </span></h2>
                   <span >Quantities available</span> <span className="bg-success px-2" > {location.state.quantity}</span>
                   <span className="pl-5"> Manufacturer :</span>     <span className="bg-info px-2"> {location.state.manufacturer}</span>
                    <h3>Product details</h3>
                    <ul>
                    <li> <p>{location.state.description}</p></li>
                    <li> <p>{location.state.description}</p></li>
                    <li> <p>{location.state.description}</p></li>
                    <li> <p>{location.state.description}</p></li>
                    <li> <p>{location.state.description}</p></li>
                    <li> <p>{location.state.description}</p></li>
                    <li> <p>{location.state.description}</p></li>
                    </ul>
                    </div>
                </div>
            </div>
            <br/>
        </div>
    
    )
}

const delFun = (id) => {
    
    const r=window.confirm("are you sure?")
    if(r){
    axios.delete(`http://localhost:5500/products/${id}`)
    .then ( d => console.log(d))
    .catch(err => console.log(err))
    toast.success('Product has been deleted successfully',{position:toast.POSITION.TOP_CENTER})
history.push('/products');

    }
    else{
       
        toast.error('You cancelled the operation',{position:toast.POSITION.TOP_CENTER})
    }
  
  }
export default withRouter(ProductDetails)
