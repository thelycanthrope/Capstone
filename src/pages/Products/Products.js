import React, { useState } from 'react'
import './card.css'
import {Link} from 'react-router-dom';
import ProductDetails from './ProductDetails'
const stt={
    
    width: "200px",
    height: "300px",
    verticalAlign: "middle",
    backgroundColor:"black"
}

   




function Products(props) {
    let path=`/products/${props.id}`;
  
    return (
        
        <div  className="bg-dark mr-5">
            
            <div className="col-sm-11 ml-3 mb-3 mt-3 a-section a-spacing-medium carddd shadow-white rounded border" id="cardd" >
                <div className="card text-center align-middle bg-dark "  >
                <Link  to={{pathname:'/products/'+props.id,state:props}} >  <img src={props.url} className="card-img-top my-2 img-thumbnail ml-2" alt="..." style={stt}/></Link>
                        <div className="card-body ">
                        <Link  to={{pathname:'/products/'+props.id,state:props}} > <h5 className="card-title text-center text-light "  id="details" style={{width:"15rem"}}>{props.pname}</h5></Link>
                            <p className="card-text text-center text-light border-bottom-0" id="price">&#8377; {props.price}</p>
                           
                        </div>
                        </div>
                </div>
            </div>
          
        )
    }

    export default Products

