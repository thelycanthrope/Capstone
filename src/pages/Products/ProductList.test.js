import React from 'react';
import {shallow,configure,mount} from 'enzyme';
import ProductsList from './ProductList';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import { BrowserRouter } from 'react-router-dom';
import Products from './Products'


configure({ adapter: new Adapter() });

describe('When valid products array props is passed to ProductsList component', () => {

    let props;
    let wrapper;
    beforeEach(() => {
        props = {
            products: [
                {
                    "pname": "SS Master 1000 English Willow Cricket Bat",
                    "description": " Lorem ipsum dolor sit amet consectetur adipisicing elit. Dicta magni et nemo, voluptate sed cum voluptates voluptatum provident",
                    "manufacturer": "Sareen Sports Industries",
                    "price": 6300,
                    "quantity": "78",
                    "url": "https://www.onlinecricstore.com/media/catalog/product/cache/26c91cbb37c57611b1daf32fd10f6d8a/u/n/untitled-1_25.png",
                    "id": 1,
                    "category": "bat"
                  },
                  {
                    "pname": "Kookaburra Kahuna 750 Wicket Keeping Gloves",
                    "quantity": 45,
                    "price": 2900,
                    "url": "https://www.onlinecricstore.com/media/catalog/product/cache/26c91cbb37c57611b1daf32fd10f6d8a/u/n/untitled-1_11_1.png",
                    "description": " Lorem ipsum dolor sit amet consectetur adipisicing elit. Dicta magni et nemo, voluptate sed cum voluptates voluptatum provident",
                    "manufacturer": "Kookaburra Sports",
                    "id": 10,
                    "category": "gloves"
                  },
                  {
                    "pname": "SG Club Leather Cricket Red Ball",
                    "quantity": 36,
                    "price": 1500,
                    "url": "https://www.onlinecricstore.com/media/catalog/product/cache/26c91cbb37c57611b1daf32fd10f6d8a/c/l/club2.jpg",
                    "description": " Lorem ipsum dolor sit amet consectetur adipisicing elit. Dicta magni et nemo, voluptate sed cum voluptates voluptatum provident",
                    "manufacturer": "SG Sports",
                    "id": 14,
                    "category": "ball"
                  }
            ]
        }
        wrapper = shallow(<ProductsList {...props}/>);
    });

    it('snapshot', () =>{
        expect(toJson(wrapper)).toMatchSnapshot();
    })
   
    it('first product name test', () => {
      
      expect(wrapper.find(Products).at(0).dive().find('#details').text()).toEqual('SS Master 1000 English Willow Cricket Bat')
      
    })
    it('2nd product price',() =>{
        expect(wrapper.find(Products).at(1).dive().find('#price').text()).toEqual('â‚¹ 2900')
    })
})
describe('When Products array props passed to ProductList Component is null', () => {
    let wrapper;
    let props;

    beforeEach(() => {
        props = {
            products: null
        }
        wrapper = shallow(<ProductsList {...props}/>);
    });

    it('should not crash and no Product component is rendered', () => {
        let product = wrapper.find('Product');
        expect(product.length).toEqual(0);
    });

});