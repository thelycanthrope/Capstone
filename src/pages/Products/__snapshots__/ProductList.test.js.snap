// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When valid products array props is passed to ProductsList component snapshot 1`] = `
<div>
  <div
    className="container "
  >
    <div
      className="row"
    >
      <Products
        category="bat"
        description=" Lorem ipsum dolor sit amet consectetur adipisicing elit. Dicta magni et nemo, voluptate sed cum voluptates voluptatum provident"
        id={1}
        key="1"
        manufacturer="Sareen Sports Industries"
        pname="SS Master 1000 English Willow Cricket Bat"
        price={6300}
        quantity="78"
        url="https://www.onlinecricstore.com/media/catalog/product/cache/26c91cbb37c57611b1daf32fd10f6d8a/u/n/untitled-1_25.png"
      />
      <Products
        category="gloves"
        description=" Lorem ipsum dolor sit amet consectetur adipisicing elit. Dicta magni et nemo, voluptate sed cum voluptates voluptatum provident"
        id={10}
        key="10"
        manufacturer="Kookaburra Sports"
        pname="Kookaburra Kahuna 750 Wicket Keeping Gloves"
        price={2900}
        quantity={45}
        url="https://www.onlinecricstore.com/media/catalog/product/cache/26c91cbb37c57611b1daf32fd10f6d8a/u/n/untitled-1_11_1.png"
      />
      <Products
        category="ball"
        description=" Lorem ipsum dolor sit amet consectetur adipisicing elit. Dicta magni et nemo, voluptate sed cum voluptates voluptatum provident"
        id={14}
        key="14"
        manufacturer="SG Sports"
        pname="SG Club Leather Cricket Red Ball"
        price={1500}
        quantity={36}
        url="https://www.onlinecricstore.com/media/catalog/product/cache/26c91cbb37c57611b1daf32fd10f6d8a/c/l/club2.jpg"
      />
    </div>
  </div>
</div>
`;
